back-end/src/main/java/com/soen343/muse/controllers/SongController.java

```diff
@@ -0,0 +1,32 @@
+package com.soen343.muse.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.soen343.muse.models.Song;
+import com.soen343.muse.repositories.SongRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping(path = "/api/songs")
+public class SongController {
+
+
+    @Autowired
+    private SongRepo songRepo;
+
+    /** return all songs */
+    @RequestMapping(value = {"", "/"}, produces = "application/json", method = RequestMethod.GET)
+    public List<Song> greeting(@RequestParam Map<String, String> params) {  
```

Comment:
Can we name this to getAllSongs or something similar?

Reply:
Fixed